app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+VALID_EMAIL_REGEX = Settings.email_regex
+class User < ApplicationRecord
+  validates :name, presence: true
+  validates :email, presence: true, length: {maximum: Settings.user.email.max_length}, format: {with: VALID_EMAIL_REGEX}, uniqueness: true
+  has_secure_password
+  before_save :downcase_email
+  private
+  def downcase_email
+    self.mail.downcase!
```

Comment:
! có tác dụng gì e nhỉ
nếu k có ! mình sẽ viết code nào thay thế

Reply:
downcase sẽ chuyển string có chữ hoa về chữ thường, nếu toàn chữ thường sẽ trả về nil
Các khác là dùng loop qua từng kí tự và kiểm tra mã ASCII

Reply:
bỏ self đi có được k e ?